//------------------------------------------------------------------------------

#include "Vector.hpp"

//------------------------------------------------------------------------------

int main()
try
{
    Vector<int> vi{0, 1, 2, 3, 4, 5 ,6};

    auto p{vi.end()};
    ++p;
    if (p != vi.end())
        cout << *p << "\n";
    /*p+=2;
    p[0];*/
    return 0;
}
catch (const exception& e) {
    cout << "\nerror: " << e.what() << '\n';
    return -1;
}
catch (...) {
    cerr << "oops!";
    return -2;
}

//------------------------------------------------------------------------------
